lib/pure/collections/heapqueue.nim

```diff
@@ -156,10 +156,12 @@ proc replace*[T](heap: var HeapQueue[T], item: T): T =
 
 proc pushpop*[T](heap: var HeapQueue[T], item: T): T =
   ## Fast version of a push followed by a pop.
-  if heap.len > 0 and heapCmp(heap[0], item):
-    swap(item, heap[0])
+  if heap.len > 0 and heapCmp(heap.data[0], item):
+    result = item
```

Comment:
```suggestion
  result = item
  if heap.len > 0 and heapCmp(heap.data[0], item):
```
along with the other suggestion (need to split them because of github)